### PR TITLE
Add metricsCounterInc into OrionServer

### PR DESCRIPTION
**Pull request by yisheng-zhou**
Add metricsCounterInc into OrionServer; use the counter inc for high alert. 

The metrics logic is as same as what is current using at orion agent: https://github.com/pinterest/orion/blob/master/orion-agent/src/main/java/com/pinterest/orion/agent/HeartbeatService.java#L85

### CODE REVIEW

File: orion-server/src/main/java/com/pinterest/orion/core/actions/alert/PagerDutyAlert.java [hunk 1]
```diff
@@ -58,6 +60,15 @@ public void initialize(Map<String, Object> config) throws PluginConfigurationExc
   @Override
   public void alert(AlertMessage message) {
     sendPager(message);
+    sendMetrics();
+  }
+
+  private void sendMetrics() {
+    try {
+      PAGERDUTY_METRICS.counter("pagerduty.alert").inc();
```
Comment (vahidhashemian) [inline orion-server/src/main/java/com/pinterest/orion/core/actions/alert/PagerDutyAlert.java:68]: How do we distinguish between different types of pagers?

Comment (yisheng-zhou) [reply orion-server/src/main/java/com/pinterest/orion/core/actions/alert/PagerDutyAlert.java:68]: This is a sample that high severity alert message is being created (paging). The title and message are both long string. I cannot put the long string into metrics path or tag directly. Removing the space to form a single word wont solve the issue properly either.

My plan is to add [AlertType] into all alert message title string. The metric sending method will grab the word inside [] and put it into metrics path. And I can change the alert message title string at different places later. 
  
```
cluster.getActionEngine().alert(AlertLevel.HIGH, new AlertMessage(
    "Serverset has less than " + (alertThreshold * 100) + "% cluster(" + cluster.getName()
        + ") nodes",
    "Serverset(" + serversetPath + ") for cluster(" + cluster.getName() + ") has less than "
        + (alertThreshold * 100) + "% of the nodes available",
    "orion"));
```

Does this sound like a plan?

File: orion-server/src/main/java/com/pinterest/orion/core/automation/operator/kafka/BrokerHealingOperator.java [hunk 2]
```diff
@@ -160,6 +161,13 @@ public void operate(KafkaCluster cluster) throws Exception {
           "Orion agents on " + cluster.getClusterId() + " are unhealthy, no URPs on the cluster: " + alertableUnhealthyAgentBrokersWithoutURPs,
           "orion"
       ));
+      OrionServer.metricsCounterInc(
+              "BrokerHealing.OrionAgent",
```
Comment (vahidhashemian) [inline orion-server/src/main/java/com/pinterest/orion/core/automation/operator/kafka/BrokerHealingOperator.java:165]: Can we use the same format for these? They ideally should have the same number of `.`'s in them.

File: orion-server/src/main/java/com/pinterest/orion/core/actions/aws/ReplaceEC2InstanceAction.java [hunk 3]
```diff
@@ -152,6 +152,13 @@ public void runAction() {
           );
           getEngine().alert(AlertLevel.MEDIUM, msg);
           getEngine().alert(AlertLevel.HIGH, msg);
+          OrionServer.metricsCounterInc(
+                  "ec2ReplacementGetState",
```
Comment (vahidhashemian) [inline orion-server/src/main/java/com/pinterest/orion/core/actions/aws/ReplaceEC2InstanceAction.java:156]: Let's clarify whether this is for old or new instance.

File: orion-server/src/main/java/com/pinterest/orion/core/actions/aws/TerminateEC2InstanceAction.java [hunk 4]
```diff
@@ -73,6 +75,13 @@ public void runAction() {
         );
         getEngine().alert(AlertLevel.MEDIUM, msg);
         getEngine().alert(AlertLevel.HIGH, msg);
```
Comment (vahidhashemian) [inline orion-server/src/main/java/com/pinterest/orion/core/actions/aws/TerminateEC2InstanceAction.java:77]: Unrelated to this diff. Wondering why there are two different alert levels here.

File: orion-server/src/main/java/com/pinterest/orion/core/actions/aws/ReplaceEC2InstanceAction.java [hunk 5]
```diff
@@ -152,6 +152,13 @@ public void runAction() {
...
+                  "failure",
```
Comment (vahidhashemian) [inline orion-server/src/main/java/com/pinterest/orion/core/actions/aws/ReplaceEC2InstanceAction.java:157]: What is the difference between `failure` and `error`? We used `error` above.

**Review by vahidhashemian — CHANGES_REQUESTED** [review]
Left a few comments.

I think it would make sense to come up with a common format for these new metrics to make them more manageable. For example,
```
orion.<subject>.<action>.<outcome>
```
e.g. `orion.broker.replacement.error`, or `orion.replacedBroker.healthCheck.timeout`....
Not sure if all can be converted to a common format like this, will leave to you to decide. The rest looks ok to me.

File: orion-server/src/main/java/com/pinterest/orion/core/actions/aws/ReplaceEC2InstanceAction.java (hunk 3 above)
Comment (yisheng-zhou) [reply orion-server/src/main/java/com/pinterest/orion/core/actions/aws/ReplaceEC2InstanceAction.java:156]: OK. Will add more context

File: orion-server/src/main/java/com/pinterest/orion/core/actions/aws/ReplaceEC2InstanceAction.java (hunk 5 above)
Comment (yisheng-zhou) [reply orion-server/src/main/java/com/pinterest/orion/core/actions/aws/ReplaceEC2InstanceAction.java:157]: If the pagerduty message uses "error", I use error. If it uses "fail", I use failure. I will make them all use error.

File: orion-server/src/main/java/com/pinterest/orion/core/actions/aws/TerminateEC2InstanceAction.java (hunk 4 above)
Comment (yisheng-zhou) [reply orion-server/src/main/java/com/pinterest/orion/core/actions/aws/TerminateEC2InstanceAction.java:77]: https://github.com/pinterest/orion/blob/master/orion-server/src/main/java/com/pinterest/orion/core/actions/alert/AlertFactory.java#L43

The alertFactory only triggers the type of alerts with same level. For example, pagerduty alert is high level and email alert is low level. It alerts twice with different levels just to trigger multiple types of alerts.

File: orion-server/src/main/java/com/pinterest/orion/core/automation/operator/kafka/BrokerHealingOperator.java (hunk 2 above)
Comment (yisheng-zhou) [reply orion-server/src/main/java/com/pinterest/orion/core/automation/operator/kafka/BrokerHealingOperator.java:165]: OK. Will update

**Review by vahidhashemian — APPROVED** [review]

**Pull request merged**